test/router/path_trie_wildcard_test.dart

```diff
@@ -0,0 +1,154 @@
+import 'package:relic/src/router/normalized_path.dart';
+import 'package:relic/src/router/path_trie.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('PathTrie Wildcard (*) Matching', () {
+    late PathTrie<int> trie;
+
+    setUp(() {
+      trie = PathTrie<int>();
+    });
+
+    test(
+        'Given a trie with path /users/*/profile, '
+        'when /users/123/profile is looked up, '
+        'then it matches with correct value and paths', () {
+      trie.add(NormalizedPath('/users/*/profile'), 1);
+      final result = trie.lookup(NormalizedPath('/users/123/profile'));
+      expect(result, isNotNull);
+      expect(result!.value, 1);
+      expect(result.parameters, isEmpty);
+      expect(result.matched.path, '/users/123/profile');
+      expect(result.remaining.segments, isEmpty);
+    });
+
+    test(
+        'Given a trie with path /*/resource, '
+        'when /any/resource is looked up, '
+        'then it matches with correct value and paths', () {
+      trie.add(NormalizedPath('/*/resource'), 1);
+      final result = trie.lookup(NormalizedPath('/any/resource'));
+      expect(result, isNotNull);
+      expect(result!.value, 1);
+      expect(result.parameters, isEmpty);
+      expect(result.matched.path, '/any/resource');
+      expect(result.remaining.segments, isEmpty);
+    });
+
+    test(
+        'Given a trie with path /files/*, '
+        'when /files/image.jpg is looked up, '
+        'then it matches with correct value and paths', () {
+      trie.add(NormalizedPath('/files/*'), 1);
+      final result = trie.lookup(NormalizedPath('/files/image.jpg'));
+      expect(result, isNotNull);
+      expect(result!.value, 1);
+      expect(result.parameters, isEmpty);
+      expect(result.matched.path, '/files/image.jpg');
+      expect(result.remaining.segments, isEmpty);
+    });
+
+    test(
+        'Given a trie with path /a/*/c/*, '
+        'when /a/b/c/d is looked up, '
+        'then it matches with correct value and paths', () {
+      trie.add(NormalizedPath('/a/*/c/*'), 1);
+      final result = trie.lookup(NormalizedPath('/a/b/c/d'));
+      expect(result, isNotNull);
+      expect(result!.value, 1);
+      expect(result.parameters, isEmpty);
+      expect(result.matched.path, '/a/b/c/d');
+      expect(result.remaining.segments, isEmpty);
+    });
+
+    test(
+        'Given a trie with path /a/*/b, '
```

Comment:
Is there a test for `/a/*` being requested with just `/a(/)` (should return null)? I don't see one.

PS: I assume trailing slashes are always optional? Or we just assume that these don't change the behavior (so that we don't have to test that variant. At least `//` is collapsed always per the normalization).

Reply:
There is:
```dart
   test(
        'Given a trie with path /a/*/b, '
        'when /a/b (fewer segments) is looked up, '
        'then no match is found', () {
      trie.add(NormalizedPath('/a/*/b'), 1);
      expect(trie.lookup(NormalizedPath('/a/b')), isNull);
    });
```
if that is what you mean.

And yes, trailing slashes will be removed by normalization.

Reply:
Well, it's a little bit different, in now the route does not match because the suffic `/b` is missing.

I was looking for a test case like

```
   trie.add(NormalizedPath('/a/*'), 1);
   expect(trie.lookup(NormalizedPath('/a')), isNull);
```